### PR TITLE
updated sample project with java version 2

### PR DESCRIPTION
**Pull request by steven-xu-lf**
- removed the usage of `CompletableFutures`
- updated both the java client core and java repo client versions to the latest java 2 SNAPSHOT versions (will update once it gets published to maven central)

### CODE REVIEW

File: src/main/java/sample/Sample.java [hunk 1]
```diff
@@ -22,51 +24,37 @@ public static void main(String[] args) {
...
-                    String creator = rootEntry.getCreator();
-                    OffsetDateTime createdDate = rootEntry.getCreationTime();
-                    System.out.printf("Root folder information:%nType: %s%nCreator: %s%nCreation Date: %s%n", entryType,
-                            creator, createdDate);
-                    return rootEntry;
-                });
+    public static Entry getRootFolder() {
+        Entry rootEntry = client.getEntriesClient().getEntry(new ParametersForGetEntry().setRepoId(config.getRepositoryId()).setEntryId(ROOT_FOLDER_ENTRY_ID).setSelect(null));
+        EntryType entryType = rootEntry.getEntryType();
+        String creator = rootEntry.getCreator();
```
Comment (jason-chan-lf) [inline src/main/java/sample/Sample.java]: The creator for the root entry seems to be emtpy string, so do we still want to print this?

File: src/main/java/sample/Sample.java [hunk 2]
```diff
@@ -22,51 +24,37 @@ public static void main(String[] args) {
...
-                .thenApply(rootEntry -> {
-                    EntryType entryType = rootEntry.getEntryType();
-                    String creator = rootEntry.getCreator();
-                    OffsetDateTime createdDate = rootEntry.getCreationTime();
-                    System.out.printf("Root folder information:%nType: %s%nCreator: %s%nCreation Date: %s%n", entryType,
-                            creator, createdDate);
-                    return rootEntry;
-                });
+    public static Entry getRootFolder() {
+        Entry rootEntry = client.getEntriesClient().getEntry(new ParametersForGetEntry().setRepoId(config.getRepositoryId()).setEntryId(ROOT_FOLDER_ENTRY_ID).setSelect(null));
```
Comment (jason-chan-lf) [inline src/main/java/sample/Sample.java]: shouldn't need this `.setSelect(null)`?

Comment (steven-xu-lf) [reply src/main/java/sample/Sample.java]: ok done, removed

File: src/main/java/sample/Sample.java (hunk 1 above)
Comment (steven-xu-lf) [reply src/main/java/sample/Sample.java]: ok done, removed

**Review by jason-chan-lf — APPROVED** [review]

**Pull request merged**